### PR TITLE
refactor(valkey_user): generate

### PR DESCRIPTION
**Pull request by byashimov**
- Migrate `aiven_valkey_user` resource and data source to the Plugin Framework
- Improve examples for array types

Resolves NEX-2760.

### CODE REVIEW

**Review by vmyroslav — DISMISSED** [review]

File: internal/plugin/service/valkey/user/zz_resource.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			},
+			"type": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Account type.",
+			},
+			"username": schema.StringAttribute{
+				MarkdownDescription: "Service username. Maximum length: `64`. Changing this property forces recreation of the resource.",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+				Required:            true,
+				Validators:          []validator.String{stringvalidator.LengthAtMost(64)},
```
Comment (ndk) [inline internal/plugin/service/valkey/user/zz_resource.go]: It seems we've missed one more validation: https://github.com/aiven/terraform-provider-aiven/pull/3005/changes#diff-b0ab00992129801f72e2523623ccf746e0b15f0344e6a92d7ed287045bb9fd21L25
```
ValidateFunc: schemautil.GetServiceUserValidateFunc(),
```

Plus, I guess we let ppl set this field as an empty string.

File: internal/plugin/service/valkey/user/user.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+func flattenACL(_ adapter.ResourceData, dto map[string]any) error {
+	accessControl, ok := dto["access_control"].(map[string]any)
+	delete(dto, "access_control")
+	if !ok {
+		return nil
+	}
+
+	for key, v := range accessControl {
+		if strings.HasPrefix(key, valkeyACLPrefix) {
```
Comment (ndk) [inline internal/plugin/service/valkey/user/user.go:87]: Looks like the field retains a stale value when the API omits it. I think it would be safe to initialize all ACL fields to empty slices first.

**Review by ndk — DISMISSED** [review]

**Comment by byashimov** [comment]
Added an extra test for `foo = []` cases where prior the provider failed to take apart `[]` and `null` (not set).
Tested locally with a Valkey instance.

**Review by ndk — APPROVED** [review]
